plugins/inputs/opcua_listener/opcua_listener_test.go

```diff
@@ -148,6 +150,148 @@ func TestSubscribeClientIntegration(t *testing.T) {
+		{"ProductUri", "0", "i", "2262", "http://open62541.org"},
+		{"ManufacturerName", "0", "i", "2263", "open62541"},
+		{"badnode", "1", "i", "1337", nil},
+		{"goodnode", "1", "s", "the.answer", int32(42)},
+		{"DateTime", "1", "i", "51037", "0001-01-01T00:00:00Z"},
+	}
+	testopctypes := []string{
+		"String",
+		"String",
+		"String",
+		"Null",
+		"Int32",
+		"DateTime",
+	}
+	testopcquality := []string{
+		"OK (0x0)",
+		"OK (0x0)",
+		"OK (0x0)",
+		"User does not have permission to perform the requested operation. StatusBadUserAccessDenied (0x801F0000)",
+		"OK (0x0)",
+		"OK (0x0)",
+	}
+	expectedopcmetrics := []telegraf.Metric{}
+	for i, x := range testopctags {
+		now := time.Now()
+		tags := map[string]string{
+			"id": fmt.Sprintf("ns=%s;%s=%s", x.Namespace, x.IdentifierType, x.Identifier),
+		}
+		fields := map[string]interface{}{
+			x.Name:     x.Want,
+			"Quality":  testopcquality[i],
+			"DataType": testopctypes[i],
+		}
+		expectedopcmetrics = append(expectedopcmetrics, metric.New("testing", tags, fields, now))
+	}
+
+	tagsRemaining := make([]string, 0, len(testopctags))
+	for i, tag := range testopctags {
+		if tag.Want != nil {
+			tagsRemaining = append(tagsRemaining, testopctags[i].Name)
+		}
+	}
+
+	subscribeConfig := SubscribeClientConfig{
+		InputClientConfig: input.InputClientConfig{
+			OpcUAClientConfig: opcua.OpcUAClientConfig{
+				Endpoint:       fmt.Sprintf("opc.tcp://%s:%s", container.Address, container.Ports[servicePort]),
+				SecurityPolicy: "None",
+				SecurityMode:   "None",
+				AuthMethod:     "Anonymous",
+				ConnectTimeout: config.Duration(10 * time.Second),
+				RequestTimeout: config.Duration(1 * time.Second),
+				Workarounds:    opcua.OpcUAWorkarounds{},
+				OptionalFields: opcua.OpcUAAdditionalFields{IncludeDataType: true},
+			},
+			MetricName: "testing",
+			RootNodes:  make([]input.NodeSettings, 0),
+			Groups:     make([]input.NodeGroupSettings, 0),
+		},
+		SubscriptionInterval: 0,
+	}
+	for _, tags := range testopctags {
+		subscribeConfig.RootNodes = append(subscribeConfig.RootNodes, MapOPCTag(tags))
+	}
+	o, err := subscribeConfig.CreateSubscribeClient(testutil.Logger{})
+	require.NoError(t, err)
+
+	// give initial setup a couple extra attempts, as on CircleCI this can be
+	// attempted to soon
+	require.Eventually(t, func() bool {
+		return o.SetupOptions() == nil
+	}, 5*time.Second, 10*time.Millisecond)
+
+	err = o.Connect()
+	require.NoError(t, err, "Connection failed")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	res, err := o.StartStreamValues(ctx)
+	require.NoError(t, err)
+
+	for {
+		select {
+		case m := <-res:
+			for fieldName, fieldValue := range m.Fields() {
+				for _, tag := range testopctags {
+					if fieldName != tag.Name {
+						continue
+					}
+					if tag.Want == nil {
+						t.Errorf("Tag: %s has value: %v", tag.Name, fieldValue)
+						return
+					}
```

Comment:
This test is not clear. What is wrong with a tag having a certain value?

Reply:
That is from the base integration test - nil-valued tags are not expected to be sent to the client, so error if the server does send one.

Reply:
I meant; please make sure the error message explains the goal. Now it just says "Tag: x has value: y".